index.js

```diff
@@ -3,14 +3,17 @@ var path = require("path");
 module.exports = function(content) {
   const defaultConfig = {
     basePath: [],
-    rewritePath: undefined
+    rewritePath: undefined,
+    emit: true
   };
 
   const config = Object.assign(defaultConfig, this.query);
   const fileName = path.basename(this.resourcePath);
 
   if (this.emitFile) {
-    this.emitFile(fileName, content, false);
+    if (config.emit) {
```

Comment:
I think that there is no point of checking for `emitFile` presence when we don't want to actually emit the file. What do you think about the following refactoring:

```js
if (this.emit) {
  if (this.emitFile) {
    this.emitFile(fileName, content, false);
  } else {
    throw new Error("emitFile function is not available");
  }
}

this.addDependency(this.resourcePath);
```